src/decorators.ts

```diff
@@ -456,6 +492,40 @@ export function DefaultArgOptional<T extends { new (...args: unknown[]) : object
    clsreg.defaultArgRequired = ArgRequiredOptions.OPTIONAL;
 }
 
+export interface InitConfigMethod {
+  new (...args: unknown[]) : object;
+  initConfiguration(ctx: InitContext, arg: unknown): Promise<void>;
+}
+export type HasInitConfigMethod<T> = T extends InitConfigMethod ? T : never;
+
+export function Configurable<T extends InitConfigMethod>() {
```

Comment:
How do methods in a class decorated with `@Configurable` interact with the HTTP endpoint decorators?

Reply:
They don't.  Configuration is not available on the handler context.

Reply:
So what happens if you decorate a method with @Workflow and also @PostApi?

Reply:
If you invoke that handler and the code calls `getConfiguredClass` it will be awarded an error: `Configuration is required for ${operationName} but was not provided.`

(This is not the first small hole in putting `@PostApi` and `@Workflow` on the same function, because replayed workflows receive most, but not all, of the same information as the original, and this error is latent.  I'm not so sure we'd advise users to put both decorators on the same function... even though for simple functions you can it is a bit of trivia involved explaining when you can and when you cannot.)